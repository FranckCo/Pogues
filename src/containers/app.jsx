import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loadUser } from 'actions/user';

import Header from 'components/header/header';
import Footer from 'components/footer/footer';

import 'scss/pogues.scss';

const mapStateToProps = state => ({
  locale: state.locale,
});

const mapDispatchToProps = {
  loadUser,
};


class AppContainer extends Component {
  static propTypes = {
    locale: PropTypes.object.isRequired,
    children: PropTypes.object.isRequired,
    loadUser: PropTypes.func.isRequired,
  };

  componentWillMount() {
    this.props.loadUser().then(user => {
      console.log('finish', user);
    });
  }

  render() {
    return (
      <div id="app">
        <Header locale={this.props.locale} />
        {this.props.children}
        <Footer locale={this.props.locale} />
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);
