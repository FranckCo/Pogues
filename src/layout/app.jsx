import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { loadUser } from 'actions/user';

import Header from 'layout/header/header';
import Footer from 'layout/footer/footer';

import { loadQuestionnaireList } from 'actions/questionnaire-list';

import 'scss/pogues.scss';

const mapStateToProps = state => {
  return {
    user: state.appState.user,
  };
};

const mapDispatchToProps = {
  loadUser,
  loadQuestionnaireList,
};

export class AppContainer extends Component {
  static propTypes = {
    children: PropTypes.object.isRequired,
    loadUser: PropTypes.func.isRequired,
    loadQuestionnaireList: PropTypes.func.isRequired,
    user: PropTypes.shape({
      name: PropTypes.string,
      permission: PropTypes.string,
      id: PropTypes.string,
      picture: PropTypes.string,
    }),
  };

  componentWillMount() {
    this.props.loadUser();
  }

  componentWillUpdate(nextProps) {
    if (nextProps.user.permission !== this.props.user.permission) {
      this.props.loadQuestionnaireList(nextProps.user.permission);
    }
  }

  render() {
    return (
      <div id="app">
        <Header />
        {this.props.children}
        <Footer />
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);
