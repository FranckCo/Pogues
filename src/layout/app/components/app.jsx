import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

import Header from 'layout/header/header';
import Footer from 'layout/footer/footer';

import 'scss/pogues.scss';

import { APP } from 'constants/dom-constants';

const { COMPONENT_ID } = APP;

// Component
const App = props => {
  useEffect(() => {
    props.loadUser();
    props.loadUnitsIfNeeded();
  }, []);

  return (
    <div id={COMPONENT_ID}>
      <Header />
      {props.children}
      <Footer />
    </div>
  );
};
// Prop types and default props

App.propTypes = {
  children: PropTypes.object.isRequired,
  loadUser: PropTypes.func.isRequired,
  loadUnitsIfNeeded: PropTypes.func.isRequired,
};
export default App;
