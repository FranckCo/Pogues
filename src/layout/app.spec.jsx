jest.dontMock('./app');
jest.mock('layout/header/header')
import React from 'react';
import renderer from 'react-test-renderer';

import { AppContainer } from './app';

describe('<App />', () => {
  const props = {
    children: <div />,
    loadUser() {},
    loadQuestionnaireList() {},
  };

  test('should have the right template', () => {
    const tree = renderer.create(<AppContainer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
