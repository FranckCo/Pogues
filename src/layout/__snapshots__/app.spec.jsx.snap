// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should have the right template 1`] = `
<div
  id="app"
>
  <div>
    Header
  </div>
  <div />
  <div />
</div>
`;
